Challenge-4/Bot_1/src/helper.ts

```diff
@@ -1,67 +1,73 @@
-import { ethers, Network } from "forta-agent";
-import { NetworkManager } from "forta-agent-tools";
+import { ethers } from "forta-agent";
 import { LRUCache } from "lru-cache";
-import { CONFIGURATOR_PROXY, CONFIGURATOR_PROXY_ARB } from "./constants";
+import { CONFIGURATOR_PROXY, CONFIGURATOR_PROXY_ARB, ERC20_ABI } from "./constants";
 
-let cache = new LRUCache<string, { [name: string]: string }>({ max: 1000 });
+export default class Helper {
+  private provider: ethers.providers.Provider;
+  private cache: LRUCache<string, { [name: string]: string }>;
 
-// Additional property for metadata field i.e. collateral values for the asset, that you can either liquidate or borrow
-export async function getCollateralAsset(
-  assetToken: string,
-  abi: string[],
-  provider: ethers.providers.Provider,
-  blockNumber: number,
-): Promise<{ [name: string]: string }> {
-  const assetInfo = new ethers.Contract(assetToken, abi, provider);
-  const key = `${assetToken}-${blockNumber}`;
-  if (cache.has(key)) {
-    return cache.get(key) as { [name: string]: string };
+  constructor(provider: ethers.providers.Provider) {
+    this.provider = provider;
+    this.cache = new LRUCache<string, { [name: string]: string }>({
+      max: 1000,
+    });
   }
-  let infoArray: { [name: string]: string } = {};
-  let infoObject: any;
-  let index = 0; // Start from 0, increment before fetching next
 
-  while (true) {
-    try {
-      infoObject = await assetInfo.getAssetInfo(index, {
-        blockTag: blockNumber,
-      });
-
-      const collateralName =
-        "Collateral Asset - " +
-        (await getCollateralName(provider, infoObject.asset, blockNumber));
-      infoArray[collateralName] = infoObject.asset;
-      index++; // Increment after successful fetch
-    } catch (error) {
-      break; // Exit loop on error
+  public async getCollateralAssets(
+    assetToken: string,
+    abi: string[],
+    blockNumber: number,
+  ): Promise<{ [name: string]: string }> {
+    const assetContract = new ethers.Contract(assetToken, abi, this.provider);
+    const key = `${assetToken}-${blockNumber}`;
+    if (this.cache.has(key)) {
+      return this.cache.get(key) as { [name: string]: string };
     }
-  }
+    let collateralAssets: { [name: string]: string } = {};
+    let index = 0; // Start from 0, increment before fetching next
 
-  cache.set(key, infoArray);
-  return infoArray;
-}
+    while (true) {
+      try {
+    let collateralAssetInfo: any;
```

Comment:
```suggestion
      try {
          let collateralAssetInfo: any;
```
The indentation of `let collateralAssetInfo: any;` is off.